### PR TITLE
Support token in provider config

### LINKED ISSUE #27: Accept Codecov API token as a provider argument
Hello and thanks for this useful tool!

Would it be possible (or would a pull request be accepted) to make this provider accept the Codecov API token as an argument rather than getting it from an environment variable?

This would enable using a Terraform reference to provide the token, like

```
provider "ucodecov" {
  token = jsondecode(data.aws_secretsmanager_secret_version.codecov_secret.secret_string)["api-token"]
}
```

...similar to how it's implemented by [`terraform-provider-github`](https://github.com/integrations/terraform-provider-github/blob/1cb5703acae3ba391229bba43f1692cfd50178df/github/provider.go#L11).

Thanks!

### PR DESCRIPTION
**Pull request by christopher-dG**
Closes #27 

I haven't actually gotten around to testing this out, but I'll do so at some point this week.

### CODE REVIEW

**Comment by christopher-dG** [comment]
The E2E tests are failing because they don't have access to `${{ secrets.CODECOV_API_TOKEN }}`.

**Comment by at-wat** [comment]
Ah yes, it doesn't work on PR from forked repo. I disabled that step on fork PR.

**Comment by christopher-dG** [comment]
Thanks! Because of the new Actions policies you'll have to press a button to run CI again :slightly_smiling_face:

File: codecov/provider.go [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
 )
 
 // Provider returns a Provider.
 func Provider() *schema.Provider {
 	provider := &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"token": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("CODECOV_API_TOKEN", nil),
```
Comment (at-wat) [inline codecov/provider.go:17]:
```suggestion
				DefaultFunc: schema.EnvDefaultFunc("CODECOV_API_TOKEN", nil),
				Sensitive: true,
```

File: codecov/provider.go [hunk 2]
```diff
@@ -1,15 +1,30 @@
...
+		},
 		DataSourcesMap: map[string]*schema.Resource{
 			"ucodecov_settings": dataSourceCodecovSettings(),
 		},
+		ConfigureContextFunc: providerConfigure,
 	}
 	return provider
 }
+
+func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
```
Comment (at-wat) [inline codecov/provider.go:29]: (for my understanding:
```go
func providerConfigure(...) (meta interface{}, _ diag.Diagnostics) {
```
)

**Review by at-wat — APPROVED** [review]
LGTM

**Comment by christopher-dG** [comment]
Awesome, thanks! Not sure what the release process is like but if you could tag a new one it would be greatly appreciated!

**Comment by at-wat** [comment]
I'll release it as v0.1.0

**Comment by at-wat** [comment]
https://github.com/at-wat/terraform-provider-ucodecov/releases/tag/v0.1.0

**Pull request merged**